test/libsolidity/SemanticTest.cpp

```diff
@@ -143,13 +143,22 @@ TestCase::TestResult SemanticTest::run(ostream& _stream, string const& _linePref
 			}
 			else
 			{
-				bytes output = test.call().useCallWithoutSignature ?
-					callLowLevel(test.call().arguments.rawBytes(), test.call().value) :
-					callContractFunctionWithValueNoEncoding(
+				bytes output;
+				if (test.call().useCallWithoutSignature)
+					output = callLowLevel(test.call().arguments.rawBytes(), test.call().value);
+				else
+				{
+					if (!m_compiler.methodIdentifiers(m_compiler.lastContractName()).isMember(test.call().signature))
```

Comment:
```suggestion
					soltestAssert(m_compiler.methodIdentifiers(m_compiler.lastContractName()).isMember(test.call().signature), "The function " + call.call().signature + " is not known to the compiler");
```
?

The exception should be caught and reported... it's not so different from the constructor assertions above, is it?

Reply:
Failure to deploy a contract cannot be attributed to a single test, while this one can.

Reply:
On the other hand an invalid constructor call *can* be attributed to a single expectation line...

I'd say everything upon which you can't just hit "update" is a ``soltestAssert``ion failure - and you can't update expectations for invalid signatures.